library/system/src/lib/yast2/etc_fstab.rb

```diff
@@ -0,0 +1,566 @@
+      }
+    end
+
+    # Parse a content line. This expects any line comment and the newline to be
+    # stripped off already.
+    #
+    # Reimplemented from ColumnConfigFile::Entry.
+    #
+    # @param line [String] content line without any line comment
+    # @param line_no [Fixnum] line number for error reporting
+    #
+    # @return [Boolean] true if success, false if error
+    #
+    # @raise [EtcFstab::ParseError] Incorrect file format
+    #
+    def parse(line, line_no = -1)
+      super
+      if @columns.size != 6
+        msg = "Wrong number of columns"
+        msg += " in line #{line_no + 1}" if line_no >= 0
+        raise EtcFstab::ParseError, msg
+      end
+      decoded_col = @columns.map { |col| EtcFstab.fstab_decode(col) }
+      @device, @mount_point, @fs_type, opt, @dump_pass, @fsck_pass = decoded_col
+      parse_mount_opts(opt)
+      @dump_pass = @dump_pass.to_i
+      @fsck_pass = @fsck_pass.to_i
+      true # success
+    end
+
+    # Populate the columns: Fill the columns with values from the other fields.
+    #
+    # This is called just prior to calculating the column widths and formatting
+    # the columns.
+    #
+    # Reimplemented from ColumnConfigFile::Entry.
+    #
+    def populate_columns
+      @columns =
+        [@device,
+         @mount_point,
+         @fs_type,
+         format_mount_opts,
+         @dump_pass.to_s,
+         @fsck_pass.to_s]
+
+      # Strictly speaking, space characters are only permitted in the
+      # mount_point column according to "man fstab". But better be safe than
+      # sorry; there might be a space character also in a device label or in
+      # the mount options, and if the kernel / the mount command / systemd can
+      # handle it, we want to support that, too; so let's simply encode all the
+      # fields.
+      #
+      # On the downside, this does not let us strip all the other fields
+      # because we might strip away a space character that was intentional; so
+      # it's the caller's responsibility to make sure that there are no
+      # unintended space characters in any of the fields.
+
+      @columns.map! { |col| EtcFstab.fstab_encode(col) }
+    end
+
+    # Get the "mount_by" type of this entry. See EtcFstab::get_mount_by.
+    #
+    # @return [Symbol]
+    #
+    # Rubocop thinks this is an accessor, but it's not
+    # rubocop:disable Style/AccessorMethodName
+    #
+    def get_mount_by
+      EtcFstab.get_mount_by(@device)
+    end
+    # rubocop:enable Style/AccessorMethodName
+
+    # Parse a mount options field and store the result in @mount_opts.
+    #
+    # This removes any occurrence of "defaults" (which really only makes sense
+    # if the field would otherwise be empty which is not permitted by the file
+    # format).
+    #
+    # @param opts [String] mount options field
+    #
+    def parse_mount_opts(opts)
+      if opts.nil?
+        @mount_opts = []
+        return
+      end
+      @mount_opts = opts.split(/,/)
```

Comment:
NP: I think simple `split(",")` should be enough here.

Reply:
From the Ruby `String::split()` reference documentation this should be the same. In what way do you think using a string rather than a regexp makes a difference here?

Reply:
Ah, no functional difference at all, just whenever I a see a regexp I always ask myself if it is actually needed.

The point is that using a simple string is more efficient (CPU and RAM) than a regexp. I do not know the Ruby internals, maybe there is some optimization done in background, but I try to avoid regexps whenever possible.

E.g. instead of `foo ~= /bar\z/` (`foo ~= /bar$/` in some cases) or you can use simple `foo.ends_with?("bar")`. No regexp compiling, etc... just a simple string matching.

The real runtime difference will be probably negligible in this case, that's why I used a "NP" (nitpicking) comment.

You can leave it as it is.